public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ArgQueryEditor/ArgQueryEditor.tsx

```diff
@@ -22,11 +23,34 @@ const ArgQueryEditor: React.FC<LogsQueryEditorProps> = ({
   onChange,
   setError,
 }) => {
+  const [subscriptions, setSubscriptions] = useState<AzureMonitorOption[]>([]);
+
+  useEffect(() => {
+    datasource.azureMonitorDatasource
+      .getSubscriptions()
+      .then((results) => {
+        const newSubscriptions = results.map((v) => ({ label: v.text, value: v.value, description: v.value }));
+        console.log('setting subscriptions', newSubscriptions);
+        setSubscriptions(newSubscriptions);
+        setError(ERROR_SOURCE, undefined);
+
+        if (!query.subscriptions || (query.subscriptions.length < 1 && newSubscriptions.length)) {
+          onChange({
+            ...query,
+            subscriptions: [query.subscription ?? newSubscriptions[0].value],
+          });
+        }
+      })
+      .catch((err) => setError(ERROR_SOURCE, err));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
lol I worry I'm a bad influence on you but also I think it might be fine

Reply:
fixed it :)